### PR TITLE
feat(delete fixtures) create fixtures for delete use case

### PR DESCRIPTION
**Pull request by samokiss**

### CODE REVIEW

File: tests/Fixtures/Classes/App/Repository/Domain/SubDomain/FunctionalEntityRepository.php [hunk 1]
```diff
@@ -16,6 +17,11 @@ public function __construct(RegistryInterface $registry)
         parent::__construct($registry, FunctionalEntityImpl::class);
     }
 
+    public function delete(FunctionalEntity $functionalEntity): void
+    {
+        $this->getEntityManager()->remove($functionalEntity);
+    }
+
     public function find($id): FunctionalEntity
```
Comment (Dzov) [inline tests/Fixtures/Classes/App/Repository/Domain/SubDomain/FunctionalEntityRepository.php]: remove ?

File: tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DTO/Request/DeleteFunctionalEntityRequestDTO.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+use OpenClassrooms\CodeGenerator\Tests\Fixtures\Classes\BusinessRules\Requestors\Domain\SubDomain\DeleteFunctionalEntityRequest;
+
+final class DeleteFunctionalEntityRequestDTO implements DeleteFunctionalEntityRequest
+{
+    /**
+     * @var int
+     */
+    public $functionalEntityId;
+
+    public function __construct(int $functionalEntityId)
```
Comment (Dzov) [inline tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DTO/Request/DeleteFunctionalEntityRequestDTO.php]: int $id

File: tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DTO/Request/DeleteFunctionalEntityRequestBuilderImpl.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public function create(): DeleteFunctionalEntityRequestBuilder
+    {
+        return $this;
+    }
+
+    public function withFunctionalEntityId(int $id): DeleteFunctionalEntityRequestBuilder
+    {
+        $this->functionalEntityId = $id;
```
Comment (killianherbunot) [inline tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DTO/Request/DeleteFunctionalEntityRequestBuilderImpl.php]: why don't you use the request as usual $this->request->functionalEntityId = $id; ?

File: tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DeleteFunctionalEntity.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace OpenClassrooms\CodeGenerator\Tests\Fixtures\Classes\BusinessRules\UseCases\Domain\SubDomain;
+
+use OpenClassrooms\CodeGenerator\Tests\Fixtures\Classes\BusinessRules\Entities\Domain\SubDomain\FunctionalEntity;
+use OpenClassrooms\CodeGenerator\Tests\Fixtures\Classes\BusinessRules\Gateways\Domain\SubDomain\FunctionalEntityGateway;
+use OpenClassrooms\CodeGenerator\Tests\Fixtures\Classes\BusinessRules\Requestors\Domain\SubDomain\DeleteFunctionalEntityRequest;
+use OpenClassrooms\UseCase\BusinessRules\Requestors\UseCaseRequest;
+
+/**
+ * @author Samuel Gomis <gomis.samuel@external.openclassrooms.com>
```
Comment (killianherbunot) [inline tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DeleteFunctionalEntity.php]: remove

File: tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/DeleteFunctionalEntityTest.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+     * @var DeleteFunctionalEntityRequestDTO
+     */
+    private $request;
+
+    /**
+     * @var DeleteFunctionalEntity
+     */
+    private $useCase;
+
+    private function buildRequest(): DeleteFunctionalEntityRequest
```
Comment (killianherbunot) [inline tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/DeleteFunctionalEntityTest.php]: I think this method should be closer to the code which call it (under setUp)

File: tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DTO/Request/DeleteFunctionalEntityRequestBuilderImpl.php (hunk 3 above)
Comment (samokiss) [reply tests/Fixtures/Classes/BusinessRules/UseCases/Domain/SubDomain/DTO/Request/DeleteFunctionalEntityRequestBuilderImpl.php]: i just do like this pr https://github.com/simpleit/SdZv4/pull/6582

File: tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/DeleteFunctionalEntityTest.php (hunk 5 above)
Comment (samokiss) [reply tests/Fixtures/Classes/Tests/BusinessRules/UseCases/Domain/SubDomain/DeleteFunctionalEntityTest.php]: yes

**Pull request merged**